src/sync/sync-utils.ts

```diff
@@ -10,6 +10,55 @@ import { SyncType, TaskType } from "~/src/sync/sync.types";
 import { sqsQueues } from "../sqs/queues";
 import { backfillFromDateToBucket } from "config/metric-helpers";
 
+export class PageSizeAwareCounterCursor {
+
+	/**
+	 *
+	 * @param pageNoOrSerialisedCursor - when a serialized cursor is provided it will scale it according to perPage value
+	 * @param perPage - ignored is plain pageNo is provided, for migration of the cursors. "Plain number" means that
+	 * 									the old sync is still in progress with pageNo size equals 20
+	 */
+	constructor(pageNoOrSerialisedCursor: string | number, perPage: number) {
+		if (!pageNoOrSerialisedCursor || Number(pageNoOrSerialisedCursor)) {
+			this.pageNo = Number(pageNoOrSerialisedCursor) || 1;
+			// Historical value we used to have in our code which was used for plain-number cursors.
+			// Ignoring perPage argument for seamless migration
+			this.perPage = 20;
+		} else {
+			const parsed = JSON.parse("" + pageNoOrSerialisedCursor) as PageSizeAwareCounterCursor;
```

Comment:
How about the graphQL cursor, which is just a string?

Reply:
This logic is only for counter-based cursor, which is PRs and Builds. The rest entity types (graphQl) should continue working as they are now

Reply:
How about extract this file? I mistakenly thought this new class is also related to sync-utils.

Reply:
Done :)